### PR TITLE
Exclude TCK from BOM

### LINKED ISSUE #1403: BOM contains unresolvable dependencies
### Issue description

BOM io.micronaut.data:micronaut-data-bom:3.3.0 (via io.micronaut:micronaut-bom:3.4.1-SNAPSHOT -> io.micronaut.data:micronaut-data-bom:3.3.0) declares a non-resolvable dependency: io.micronaut.data:micronaut-data-tck:3.3.0
BOM io.micronaut.data:micronaut-data-bom:3.3.0 (via io.micronaut:micronaut-bom:3.4.1-SNAPSHOT -> io.micronaut.data:micronaut-data-bom:3.3.0) declares a non-resolvable dependency: io.micronaut.data:micronaut-data-document-tck:3.3.0

### PR DESCRIPTION
**Pull request by timyates**
Fixes #1403

### CODE REVIEW

**Review by sdelamo — APPROVED** [review]

**Comment by dstepanov** [comment]
Please add `data-document-tck`

File: data-bom/build.gradle [hunk 1]
```diff
@@ -7,6 +7,6 @@ version projectVersion
 
 micronautBom {
     excludeProject.set({ p ->
-        p.name.contains('benchmark') || p.name.contains('example')
+        p.name.contains('benchmark') || p.name.contains('example') || p.name == 'data-tck'
```
Comment (sdelamo) [inline data-bom/build.gradle:10]: Based on @dstepanov comment I think we should exclude with ` || p.name.endsWith('-tck')`

**Review by sdelamo — CHANGES_REQUESTED** [review]
Please add `data-document-tck

**Comment by timyates** [comment]
Done.  Sorry about that, I missed the other one 😔

**Review by sdelamo — APPROVED** [review]

**Comment by sdelamo** [comment]
@dstepanov please merge it and release a patch version once CI passes. 

I will merge the Core PR removing the suppression we added to `bom/build.gradle` in core.

**Pull request merged**